src/main/java/com/epam/aidial/core/service/ResourceService.java

```diff
@@ -208,6 +209,16 @@ public ResourceItemMetadata putResource(ResourceDescription descriptor, String b
         }
     }
 
+    public void computeResource(ResourceDescription descriptor, Function<String, String> fn) {
+        String redisKey = redisKey(descriptor);
+
+        try (var ignore = lockService.lock(redisKey)) {
```

Comment:
The lock is not reetrant. Please use real server and service in the tests.

Reply:
fixed